test/test_files/function/gds/basic.test

```diff
@@ -114,3 +114,13 @@ Hubert Blaine Wolfeschlegelsteinhausenbergerdorff|0.018750
 [Alice,Dan,Bob]|[2021-06-30,1950-05-14]|[0:0,0:3]|[0:3,0:1]|Alice|Bob
 [Alice,Dan,Carol]|[2021-06-30,2000-01-01]|[0:0,0:3]|[0:3,0:2]|Alice|Carol
 [Alice,Dan]|[2021-06-30]|[0:0]|[0:3]|Alice|Dan
+-STATEMENT MATCH p = (a:person)-[e:knows* 3..3]->(b:person) WHERE a.ID = 0 RETURN COUNT(*)
```

Comment:
One edge case I'm curious about is whether the source and destination nodes are the same for acyclic. I believe such paths are also not allowed, right? To be more specific, I'm curious about what to do in the following query:
```
MATCH (a:Person)-[r:Knows* ACYCLIC]->(b:Person)
WHERE a.name = Alice and b.name=Alice
RETURN *
```
So suppose there is an Alice node, node 5, and node 5 has a self loop edge to itself. Does that qualify as a valid output? Or say an edge like node 5 -[Knows]-> node 7 -[Know]-> node 5. The question is the [r:Knows*] part of this matched pattern only involves -[Knows]-> node 7 -[Know]-, which is acyclic (though is not really a path), but if you consider the source and end nodes, the full path, i.e., p, is acyclic.  I would learn what should be done here (maybe look at Neo4j) and then add a test case for it.

Reply:
Neo4j doesn't have this ACYCLIC keyword in the recursive pattern.

```
MATCH p = (a:Person)-[r:Knows* ACYCLIC]->(b:Person)
WHERE a.name = Alice and b.name=Alice
RETURN *
```

In your example, `r` is acyclic as it only involves `-[Knows]-> node 7 -[Know]-`. But `p` is cyclic.